src/libclipper/include/clipper/threadpool.hpp

```diff
@@ -0,0 +1,269 @@
+#ifndef CLIPPER_LIB_THREADPOOL_HPP
```

Comment:
Is there a source from which this threadpool was modified, or did you roll your own? If applicable, we should reference it here, and it would be helpful to see what was changed.